src/hooks/useRecentURLs.ts

```diff
@@ -0,0 +1,33 @@
+import { useLocalStorage } from 'react-use'
+
+const LOCAL_STORAGE_KEY = 'recentURLs'
+const MAX_RECENT_URLS = 3
+
+export function useRecentURLs() {
+  const [recentURLs = [], setRecentURLs] = useLocalStorage<string[]>(
+    LOCAL_STORAGE_KEY,
+    []
+  )
+
+  const addURL = (url: string) => {
+    const trimmedURL = url.trim()
```

Comment:
As an improvement, we should consider normalizing the URL to lowercase to avoid issues with case sensitivity.

<img width="276" alt="image" src="https://github.com/user-attachments/assets/cc1cb77d-b3e9-4cf4-ad19-0cbafc58830d" />